core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1085,6 +1085,10 @@ class DAGScheduler(
 
         if (disallowStageRetryForTest) {
           abortStage(failedStage, "Fetch failure will not retry stage due to testing config")
+        } else if (failedStage.failAndShouldAbort()) {
```

Comment:
Yup exactly! You might wait until @squito gives the thumb's up before implementing that though -- just to make sure I haven't missed something, because from his comment it sounded like he didn't think that would work.